src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java

```diff
@@ -622,6 +622,11 @@ protected Query doToQuery(QueryShardContext context) {
                     String.format(Locale.ROOT, "Engine [%s] does not support radial search", knnEngine)
                 );
             }
+            if (vectorDataType == VectorDataType.BINARY) {
+                throw new UnsupportedOperationException(
+                    String.format(Locale.ROOT, "Binary data type does not support radial search", knnEngine)
```

Comment:
knnEngine is not getting added in the exception message. You missed setting %s in the messsage,